utils/mgmt_system/google.py

```diff
@@ -0,0 +1,290 @@
+# coding: utf-8
+"""Backend management system classes
+
+Used to communicate with providers without using CFME facilities
+"""
+
+from oauth2client.client import flow_from_clientsecrets
+from oauth2client.file import Storage
+from oauth2client.tools import run_flow, argparser
+from googleapiclient.discovery import build
+from utils.mgmt_system.base import MgmtSystemAPIBase, VMInfo
+from utils.mgmt_system.exceptions import VMInstanceNotFound
+from utils.log import logger
+from utils.wait import wait_for
+
+
+class GoogleCloudSystem (MgmtSystemAPIBase):
+    """
+    Client to Google Cloud Platform API
+
+    """
+
+    states = {
+        'running': ('RUNNING',),
+        'stopped': ('TERMINATED',),
+        'starting': ('STAGING'),
+        'stopping': ('STOPPING'),
+    }
+
+    def __init__(self, project=None, zone=None, default_instance_name=None, scope=None,
+            oauth2_storage=None, client_secrets=None):
+
+        self._project = project
+        self._zone = zone
+
+        # Default name of the instance name from config file
+        self._instance_name = default_instance_name
+
+        # Perform OAuth 2.0 authorization.
+        # based on OAuth 2.0 client IDs credentials from client_secretes file
+        if client_secrets and scope and oauth2_storage:
+            flow = flow_from_clientsecrets(client_secrets, scope=scope)
+            storage = Storage(oauth2_storage)
+            self._credentials = storage.get()
+
+        if self._credentials is None or self._credentials.invalid:
+            self._credentials = run_flow(flow, storage, argparser.parse_args([]))
+
+        if self._credentials is None or self._credentials.invalid:
+            raise Exception("Incorrect credentials for Google Cloud System")
+
+        self._compute = build('compute', 'v1', credentials=self._credentials)
+
+    def _get_all_instances(self):
+        return self._compute.instances().list(project=self._project, zone=self._zone).execute()
+
+    def list_vm(self):
+        result = self._get_all_instances()
+        return result['items']
```

Comment:
The function returned the list of dictionaries with properties of vm, including name. 
Modified it to return just the list of names and in case there is no instances - empty list.